src/mesh.cpp

```diff
@@ -1,80 +1,118 @@
 #include <boost/filesystem/operations.hpp>
 #include <mesh.hpp>
 
+#include <algorithm>
 #include <boost/algorithm/string.hpp>
 #include <fstream>
 #include <iostream>
 #include <limits>
 #include <sstream>
 
+#include <vtkDoubleArray.h>
+#include <vtkGenericDataObjectReader.h>
+#include <vtkPointData.h>
+#include <vtkPoints.h>
+#include <vtkSmartPointer.h>
+#include <vtkUnstructuredGrid.h>
+#include <vtkUnstructuredGridWriter.h>
+
 namespace aste {
 
 // --- MeshName
 
 std::string MeshName::filename() const
 {
-  return _mname + ".txt";
+  return _mname + ".vtk";
+}
+
+void MeshName::setDataname(std::string dataname)
+{
+  _dname = dataname;
 }
 
-std::string MeshName::connectivityfilename() const
+std::string MeshName::dataname() const
 {
-  return _mname + ".conn.txt";
+  return _dname;
 }
 
 namespace {
 // Reads the main file containing the vertices and data
-void readMainFile(Mesh &mesh, const std::string &filename)
+void readMainFile(Mesh &mesh, const std::string &filename, const std::string &dataname, const int &dim)
 {
+
   if (!fs::is_regular_file(filename)) {
     throw std::invalid_argument{"The mesh file does not exist: " + filename};
   }
-  std::ifstream mainFile{filename};
-  std::string   line;
-  while (std::getline(mainFile, line)) {
-    double             x, y, z, val;
-    std::istringstream iss(line);
-    iss >> x >> y >> z >> val; // split up by whitespace
-    std::array<double, 3> vertexPos{x, y, z};
-    mesh.positions.push_back(vertexPos);
-    mesh.data.push_back(val);
-  }
-}
 
-// Reads the connectivity file containing the triangle and edge information
-void readConnFile(Mesh &mesh, const std::string &filename)
-{
-  if (!fs::is_regular_file(filename)) {
-    throw std::invalid_argument{"The mesh connectivity file does not exist: " + filename};
+  vtkSmartPointer<vtkGenericDataObjectReader> reader =
+      vtkSmartPointer<vtkGenericDataObjectReader>::New();
```

Comment:
I think it is the style of the library since almost every method returns a pointer. 

using it directly will result such weird code pieces

```
vtkPoints *Points    = reader.GetUnstructuredGridOutput()->GetPoints();
```